### PR TITLE
added test cases for verifying color contrast analyzer

### PR DESCRIPTION
**Pull request by babylonbee**
#### PR checklist

- [ ] Run through of all [test scenarios](https://github.com/Microsoft/accessibility-insights-windows/blob/master/docs/Scenarios.md) completed?
- [[144]](https://github.com/Microsoft/accessibility-insights-windows/issues/144) Does this address an existing issue? If yes, Issue# - 
- [ ] Includes UI changes?
  - [ ] Run the production version of Accessibility Insights for Windows against a version with changes.
  - [x] Attach any screenshots / GIF's that are applicable.

simple_black_and_white_title

simple_grey_and_white_title

simple_purple_and_white_button

simple_white_and_black_text

simple_blue_and_white_text

Note - After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge. 

#### Describe the change
This PR is to add more test images and test cases to verify the color contrast algorithm made by @chriscm2006 , which is related to [issue144](https://github.com/Microsoft/accessibility-insights-windows/issues/144). [This ](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMicrosoft%2Faccessibility-insights-windows%2Fblob%2Fdequelabs%2Fcolor_contrast%2Fdocs%2FAddingColorContrastTests.md&data=02%7C01%7CBi.Wu%40microsoft.com%7C7e9a34a8d1164c14d4d108d6ae4934c0%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636888026772600701&sdata=9LDyVBt0c271RqaKyCRc8G44psGpqWCNGSImsJm4%2BY4%3D&reserved=0)is the related instruction to create the tests.

### CODE REVIEW

**Comment by chriscm2006** [comment]
I generally approve. This is perfect. I would like to wait until my PR is merged OR you could make this PR against the dequelabs/colorcontrast branch instead, and if it causes problems I can fix before we merge that in...

In that case I would approve right away.

I would talk with @neocln before picking a course of action.

**Comment by babylonbee** [comment]
> I generally approve. This is perfect. I would like to wait until my PR is merged OR you could make this PR against the dequelabs/colorcontrast branch instead, and if it causes problems I can fix before we merge that in...
> 
> In that case I would approve right away.
> 
> I would talk with @neocln before picking a course of action.

Thanks Chirs. I reached out to @neocln, if you are okay, let's hold on merging this PR until your PR is merged into Master. So feel free to merge your PR once it is ready.

**Comment by chriscm2006** [comment]
This looks good, assuming the conflicts get resolved! :) I'll go ahead and check the box, if it gets unchecked when you fix the conflicts let me know!

**Comment by babylonbee** [comment]
> This looks good, assuming the conflicts get resolved! :) I'll go ahead and check the box, if it gets unchecked when you fix the conflicts let me know!

Thanks @chriscm2006! I fixed the conflict. The algorithm now is more accurate in calculating the color so I updated the test cases as well. Apart from that, there are 2 failing test cases, it looks like the mostContrastingPair value calculated from RunColorContrastCalculation() is **null** for some .bmp files, namely, simple_blue_and_white_text.bmp and simple_purple_and_white_button.bmp. Could you please double check them and let us know what we can do for them?

**Comment by chriscm2006** [comment]
@babylonbee is there an issue for this? There is no current logic problem, this is a heuristics problem. We can play with the heuristics. I would love to have an issue to reference.

As per this pull request, this looks great!

**Comment by babylonbee** [comment]
> @babylonbee is there an issue for this? There is no current logic problem, this is a heuristics problem. We can play with the heuristics. I would love to have an issue to reference.
> 
> As per this pull request, this looks great!

Yep, I created the related issue: https://github.com/Microsoft/accessibility-insights-windows/issues/231

File: src/AccessibilityInsights.DesktopTests/ColorContrastAnalyzer/ImageTests.cs [hunk 1]
```diff
@@ -38,6 +38,73 @@ public void SimpleBlackAndGreyButton()
             Assert.AreEqual(Confidence.High, result.ConfidenceValue());
         }
 
+        // TODO: we will revisit this test case after issue#231 has been fixed
```
Comment (DaveTryon) [inline src/AccessibilityInsights.DesktopTests/ColorContrastAnalyzer/ImageTests.cs]:
```suggestion
        // [TestMethod]    TODO: we will revisit this test case after issue#231 has been fixed
```

File: src/AccessibilityInsights.DesktopTests/ColorContrastAnalyzer/ImageTests.cs [hunk 2]
```diff
@@ -38,6 +38,73 @@ public void SimpleBlackAndGreyButton()
...
+
+            var result = image.RunColorContrastCalculation();
+
+            Assert.AreEqual(new ColorPair(new CCColor(55, 0, 110), new CCColor(255, 255, 255)),
+                result.GetMostLikelyColorPair());
+
+            Assert.AreEqual(Confidence.Low, result.ConfidenceValue());
+        }
+
+        // TODO: we will revisit this test case after issue#231 has been fixed
```
Comment (DaveTryon) [inline src/AccessibilityInsights.DesktopTests/ColorContrastAnalyzer/ImageTests.cs]:
```suggestion
        // [TestMethod]    TODO: we will revisit this test case after issue#231 has been fixed
```

**Comment by babylonbee** [comment]
> ome other

This PR is to add more test cases to further support the color contrast algorithm. It addresses the issue#144 by adding the image input and verify the constricting pair colors match those returned by the color contrast algorithm.

**Pull request merged**